node/core/dispute-coordinator/src/lib.rs

```diff
@@ -312,6 +313,11 @@ async fn handle_incoming(
 				session,
 				statements,
 			).await?;
+			// TODO: Recover availability and report back unavailable data:
+			// https://github.com/paritytech/polkadot/issues/3399
```

Comment:
Dispute participation recovers availability. I assume we should not recover availability on every single imported statement, but only on those that initiate participation.